StripePaymentSheet/StripePaymentSheet/Source/PaymentSheet/New Payment Method Screen/WalletHeaderView.swift

```diff
@@ -152,14 +163,37 @@ extension PaymentSheetViewController {
 
             if let lastButton = buttons.last {
                 stackView.setCustomSpacing(Constants.labelSpacing, after: lastButton)
+                isApplePayLastButton = lastButton == applePayButton
             }
 
             addAndPinSubview(stackView)
         }
+        private func updateStackView() {
+            // Update Apple Pay Button's color
+            let isBlackApplePayButton = appearance.colors.background.contrastingColor == .black
+            if isBlackApplePayButton {
+                if let whiteApplePayButtonIndex = stackView.arrangedSubviews.firstIndex(of: applePayButtonWhite) {
+                    applePayButtonWhite.removeFromSuperview()
+                    stackView.removeArrangedSubview(applePayButtonWhite)
+                    stackView.insertArrangedSubview(applePayButtonBlack, at: whiteApplePayButtonIndex)
+                    if isApplePayLastButton {
+                        stackView.setCustomSpacing(Constants.labelSpacing, after: applePayButtonBlack)
+                    }
+                }
+            } else {
+                if let blackApplePayButonIndex = stackView.arrangedSubviews.firstIndex(of: applePayButtonBlack) {
+                    applePayButtonBlack.removeFromSuperview()
+                    stackView.removeArrangedSubview(applePayButtonBlack)
+                    stackView.insertArrangedSubview(applePayButtonWhite, at: blackApplePayButonIndex)
+                    if isApplePayLastButton {
+                        stackView.setCustomSpacing(Constants.labelSpacing, after: applePayButtonWhite)
+                    }
+                }
```

Comment:
Seems like we could simplify by adding both white and black and then hide one or the other. But NBD, feel free to ship as-is.